telethon/telegram_client.py

```diff
@@ -297,7 +297,14 @@ def code_callback():
                              'must only provide one of either')
 
         if not self.is_connected():
-            self.connect()
+            connected = self.connect()
+            while connected is not True:
```

Comment:
Yet you never update `connected` again. That's a strange way to do it. Might as well use a `while True` directly.